content/cloud/deployment-concepts.md

```diff
@@ -8,54 +8,46 @@ url = "https://github.com/fermyon/developer/blob/main/content/cloud/deployment-c
 
-Let's unfold each of these steps.
+> Note: The Fermyon Cloud Spin plugin supports upgrading an application which was deployed using an version older than 0.3.0. However, you cannot use a version prior to 0.3.0 to upgrade an application, which was already deployed using 0.3.0.
 
-### 1. Packaging and Uploading an Application
-
-The first step in deploying an application is to package all the files into parcels and generate an invoice.  All of this is handled in a staging directory, which is either a temporary directory using [this implementation](https://doc.rust-lang.org/std/env/fn.temp_dir.html#platform-specific-behavior) or the staging directory defined using the `--staging-dir` option.
-
-The bindle will be named using the `name` from `spin.toml`, the `version` from `spin.toml`, and a build metadata string, which is automatically generated by Spin at deployment time.
-
-Bindles in the Fermyon Cloud always use Semantic Versioning and require major, minor, and patch version numbers to be specified. Therefore, the version in `spin.toml` has to conform to the `MAJOR.MINOR.PATCH` format, i.e. `1.0.1` is valid, `1.0` and `1` are not valid version numbers for a Spin application. The result is that the version of a Spin application will be like this `my_app_name/1.0.0+r80e5abb`.
+## The Deployment Process Explained
 
-Following packaging, the bindle will be uploaded to the Fermyon Cloud. As soon as a bindle has been uploaded, it cannot be modified or deleted. This is to preserve the integrity of the immutability of bindles.
+When you run the `spin cloud deploy` command, the following happens:
 
-### 2. Create or Upgrade an Application
+1. Packaging the application and uploading it to the Fermyon Cloud
+  - The first step in deploying an application is to package all the files using the OCI specification. The Cloud Plugin will create individual layers for the files that make up the application, to optimize changes to applications on subsequent packaging and deployment.
+  - Following packaging, the image will be uploaded to an OCI compatible registry in the Fermyon Cloud.
+2. Creating or upgrading an application
+  - An application is upgraded if the application name, as defined in `spin.toml`, already exists in your Fermyon Cloud account.
+  - Before serving traffic to the application, the deployment process checks for the application health endpoint and finishes once the application is concluded to be healthy by the cloud. The application health point is an integral part of the Fermyon Cloud, which does reserve the HTTP route `/.well-known/spin/health`, which will not be routed to your Spin components.
+  - If an upgrade takes place, the failover from the old to the new deployment takes a short amount of time, during which you can observe replies from both deployments (version).
 
-The next step in the deployment process is to create or upgrade the application.
+## Deprecating Bindle - An Aggregate Object Storage System
```

Comment:
Do we want to add a specific section here or elsewhere in this doc around the (hopefully rare) case when a user may re-deploy an OCI-based app using Bindle (eg cloud plugin <= 0.2.0)?  An error will occur on `spin deploy` and we will link to this article, so having a dedicated entry on why the error is occuring and what the user can do to avoid it could be useful.  Heading might be "Cannot switch deployment method from OCI to Bindle" or could focus on cloud plugin versions.  (eg if app is deployed with >= 0.3.0 cloud plugin, it can't be redeployed with an earlier version).

Reply:
I kept this somewhat vague, but also clear in that the cloud will soon stop supporting Bindle, so get on the latest train :-)